### ISSUE TITLE
gnt-network info [name] very slow for /16 networks

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1072]
Originally reported of Google Code with ID 1016.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

<b>What distribution are you using?</b>
wheezy-backports with ganeti 2.11.6 from repo

<b>What steps will reproduce the problem?</b>
1. gnt-network add --network=10.42.0.0/16 --gateway=10.42.0.1 [...]
2. gnt-network info [name]

<b>What is the expected output? What do you see instead?</b>
[...]
       0 XX..............................X..............................X 63
[...]
     65472 ...............................................................X 65535
[...]


<b>Please provide any additional information below.</b>
would it be possible to simply not display the the map if the network is larger than /24?

Notice: as I'm parsing the network settings to configure the VM I can't really use a smaller Network without switching to routed and adding a gateway.
```
Originally added on 2015-01-05 23:47:31 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This is caused by Issue 397 basically. Filing it as duplicate.
```
Originally added on 2015-01-07 07:53:05 +0000 UTC.

Mentioned Issue with GoogleCode ID 397 has been migrated to GitHub with ID #453.
Changed State: Duplicate
Marked as duplicate of #453

**Comment by Ganeti-Issues-Migrator** [comment]
```
I don't think this has to do with Issue 397. If you use GanetiRapiClient.GetNetwork("network-with-16-netmask") the network info is returned really fast. The reason why gnt-network info takes so long is textrawp module used
for pretty printing.
```
Originally added on 2015-01-07 15:34:07 +0000 UTC.

Mentioned Issue with GoogleCode ID 397 has been migrated to GitHub with ID #453.
Added Labels: Priority-Low Component-networking
Changed State: Accepted
No longer marked as duplicate.{false 1016 I don't think this has to do with Issue 397. If you use GanetiRapiClient.GetNetwork("network-with-16-netmask") the network info is returned really fast. The reason why gnt-network info takes so long is textrawp module used
for pretty printing. 2015-01-07 15:34:07 +0000 UTC 0 [] map[397:453] [Priority-Low Component-networking] [] Accepted   map[] map[] map[] map[] 0 true}

**Comment by Ganeti-Issues-Migrator** [comment]
```
Ah sorry, then I mistook it. Thanks for un-duplicating it.
```
Originally added on 2015-01-07 16:13:24 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Reproduced. As far as I understand the bug report, the request is to
no display the usage map for networks larger than /24.

As this is an incompatible (albeit reasonable) change, the fix should
go to the master branch.

Will see what I can do here.
```
Originally added on 2015-04-08 14:13:30 +0000 UTC.

Added Labels: Type-Enhancement
Added to Milestone: Release2.15

**Comment by Ganeti-Issues-Migrator** [comment]
```
OK, found the source of slowness (textwrap.wrap). So I think
we can leave the semantics of gnt-network info as is and just
speed things up. So retargeting for 2.11. Currently working on
a fix.
```
Originally added on 2015-04-09 07:54:26 +0000 UTC.

Changed State: Started
Added to Milestone: Release2.11
Removed from Milestone: Release2.15

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 40f3aa6fb02e0c1addc0479925def815caec0268
Author: Klaus Aehlig <aehlig@google.com>
Date:   Thu Apr 9 10:20:37 2015 +0200

    Replace textwrapper.wrap by a custom version for networks
    
    "gnt-network info" also provides information about the mapping of used
    IPs in that network. This mapping is obtained from the query as a long
    string without any spaces. It seems that textwrap.wrap does not perform
    well on those strings. Therefore, manually wrap that string, thus bringing
    down the time "gnt-network info" takes for a /16 from 1.5 minutes to half
    a second. Fixes issue 1016.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2015-04-09 10:29:46 +0000 UTC.

Mentioned Issue with GoogleCode ID 1016 has been migrated to GitHub with ID #1072.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:05 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:42 +0000 UTC.

Changed State: Released
